core/services/gateway/handlers/capabilities/handler.go

```diff
@@ -177,12 +178,72 @@ func (h *handler) handleWebAPITriggerMessage(ctx context.Context, msg *api.Messa
 	return nil
 }
 
+func (h *handler) handleWebAPIOutgoingMessage(ctx context.Context, msg *api.Message, nodeAddr string) error {
+	h.lggr.Debugw("handling webAPI outgoing message", "messageId", msg.Body.MessageId, "nodeAddr", nodeAddr)
+	if !h.nodeRateLimiter.Allow(nodeAddr) {
+		return fmt.Errorf("rate limit exceeded for node %s", nodeAddr)
+	}
+	var payload Request
+	err := json.Unmarshal(msg.Body.Payload, &payload)
+	if err != nil {
+		return err
+	}
+
+	timeout := time.Duration(payload.TimeoutMs) * time.Millisecond
+	req := network.HTTPRequest{
+		Method:  payload.Method,
+		URL:     payload.URL,
+		Headers: payload.Headers,
+		Body:    payload.Body,
+		Timeout: timeout,
+	}
+
+	// send response to node async
+	h.wg.Add(1)
+	go func() {
+		defer h.wg.Done()
+		// not cancelled when parent is cancelled to ensure the goroutine can finish
```

Comment:
how will the handler shutdown cleanly if it is waiting for the timeout of the context of this routine to end? parents should always reserve the ability to shutdown a child routine even if the contexts are separated.

Reply:
this is actually a rename of the existing `handleWebAPITargetMessage` (that I forgot to clean 🧹 ) will delete this. 
We can probably we can get Davids input on this and tackle it in a separate PR, given the comment it seems we intentionally want to interrupt the goroutine even if the parent is cancelled.